stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -110,6 +110,30 @@ for (t1, t2) in ((:UnitUpperTriangular, :UpperTriangular),
     end
 end
 
+# Adding a complex UniformScaling to the diagonal of a Hermitian
+# matrix breaks the hermiticity, if the UniformScaling is non-real.
+# However, to preserve type stability, we do not special-case a
+# UniformScaling{<:Complex} that happens to be real.
+function (+)(A::Hermitian, J::UniformScaling{<:Complex})
```

Comment:
Couldn't this just be
```suggestion
(+)(A::Hermitian{T,S}, J::UniformScaling{<:Complex}) where {T,S} = S(A) + J
```

Reply:
That would result in two copies though. Maybe `... = (R = S(A); R .+= J; R)`.

Reply:
Or I guess
```
R = S(A); R[diagind(R)] .+= I.λ; R
```
since we can't broadcast `UniformScaling`.

Reply:
@andreasnoack @fredrikekre  I don't think this would be an ideal solution.  We need to promote to a complex type before adding to the diagonal, so we would have to special case real Hermitian matrices.  But the `copytri!` may be unnecessary, it was just the fastest conversion I could find.

Reply:
Actually, the tests fail for `S(A)` when `S` is a `SubArray`, so `copytri!` is probably necessary here.

Reply:
Hm. This was what `full` used to do and we don't have a complete replacement. It was assumed that using type parameters or explicit constructor would be sufficient but your example is a case where a generic function would be better.